apps/chef_objects/src/chef_user.erl

```diff
@@ -347,8 +363,12 @@ update_from_ejson(#chef_user{} = User, UserEJson) ->
     end,
     Name = username_from_ejson(UserEJson),
     Email = ej:get({<<"email">>}, UserEJson),
-    ExternalAuthenticationUid = value_or_null({<<"external_authentication_uid">>}, UserEJson),
-    RecoveryAuthenticationEnabled = ej:get({<<"recovery_authentication_enabled">>}, UserEJson) =:= true,
```

Comment:
Interestingly, I *think* this behavior is present in opscode-account for both fields as well (I did a quick search and didn't see anything to preserve it on update). 

In the case of external_authentication_uid this change makes sense.  However, in the case of recovery_authentication_enabled, I don't know that it's correct: we don't provide any other means of resetting this value once someone logs in.  Given that, the change may be to preserve it here as you've done, but to modify authenticate_user endpoint to automatically reset it upon successful login?  This could be handled in oc_chef_wm_authenticate_user:maybe_upgrade_password, which already potentially updates the user record at time of successful auth.